user-interface/src/main/java/life/qbic/datamanager/views/notifications/Toast.java

```diff
@@ -0,0 +1,248 @@
+  }
+
+  @Override
+  public void open() {
+    var cssClass = switch (type) {
+      case SUCCESS -> "success-toast";
+      case INFO -> "info-toast";
+    };
+    addClassName(cssClass);
+    super.open();
+  }
+
+  private void setType(Type type) {
+    this.type = requireNonNull(type, "type must not be null");
+  }
+
+
+  /**
+   * Creates a toast showing the text provided.
+   *
+   * @param text
+   * @return
+   */
+  public static Toast createWithText(String text) {
+    Div textDiv = new Div();
+    textDiv.addClassName("text-content");
+    textDiv.setText(text);
+    return create(textDiv);
+  }
+
+  /**
+   * Creates a toast showing the component.
+   *
+   * @param content
+   * @return
+   */
+  public static Toast create(Component content) {
+    requireNonNull(content, "content must not be null");
+    Toast toast = new Toast();
+    toast.withContent(content);
+    return toast;
+  }
+
+  /**
+   * Expands the toast and includes a link to a specific route target.
+   *
+   * @param linkText         The text of the link shown to the user
+   * @param navigationTarget the target of the navigation
+   * @param routeParameters  the parameters used for navigation
+   * @return
+   */
+  public Toast withRouting(String linkText, Class<? extends Component> navigationTarget,
+      RouteParameters routeParameters) {
+    return this.withContent(Toast.createRoutingContent(this.content, createRoutingComponent(
+        linkText, navigationTarget, routeParameters)));
+  }
+
+  /**
+   * Sets the content of the toast
+   *
+   * @param content the content of the toast to set.
+   * @return
+   */
+  protected Toast withContent(Component content) {
+    requireNonNull(content, "content must not be null");
+    //we only want to remove the current content from its parent if the parent is not the new content.
+    //Otherwise the current content would be removed from the new content as well.
+    if (nonNull(this.content) && !ComponentFunctions.isParentOf(content, this.content)) {
+      this.content.removeFromParent();
+    }
+    this.content = content;
+    this.content.addClassName("toast-content");
+    refresh();
+    return this;
+  }
+
+  private void refresh() {
+    removeAll();
+    add(this.content, this.closeButton);
+  }
+
+  /**
+   * Can this toast be closed by the user?
+   *
+   * @return
+   */
+  public boolean isCloseable() {
+    return closeable;
+  }
+
+  /**
+   * Specifies whether the toast includes a button for the user to close the toast.
+   *
+   * @param closeable
+   * @return
+   */
+  public Toast setCloseable(boolean closeable) {
+    closeButton.setVisible(closeable);
+    this.closeable = closeable;
+    return this;
+  }
+
+  /**
+   * Creates a matching toast content containing routing components with the correct css classes.
+   *
+   * @param content
+   * @param routingComponent
+   * @return
+   */
+  public static Component createRoutingContent(Component content, Component routingComponent) {
```

Comment:
I think this API is way out of scope for this class. Moreover, not a Toast is created, but a component.